### PR TITLE
Added unset-* label

### PR DESCRIPTION
**Pull request by clecat**
Added a unset-* label, allowing to remove an environment variable in a shell block.

It works the same way as set-* expect that it does not need a value:
>  '''sh unset-HOME
>    $ echo $HOME
>
> '''

It's original purpose was to allow shell script using Dune to run normally:
Dune sets a variable INSIDE_DUNE to 1 when running, which implies some modified behaviours.

So in the case of mdx being used by a dune rule, and itself launching dune in an sh block, the variable is defined before running the nested dune.

This is why, after talking about it with @samoht , I added INSIDE_DUNE to the blacklist by default.

### CODE REVIEW

File: lib/block.ml [hunk 1]
```diff
@@ -141,15 +143,16 @@ let pp ?syntax ppf b =
...
-  `Label "dir"              , [`Any];
-  `Label "source-tree"      , [`Any];
-  `Label "file"             , [`Any];
-  `Label "part"             , [`Any];
-  `Label "env"              , [`Any];
-  `Label "skip"             , [`None];
-  `Label "non-deterministic", [`None; `Some "command"; `Some "output"];
-  `Label "version"          , [`Any];
-  `Prefix "set-"            , [`Any];
+  {s = "dir"              ; t = `Label ; vs = [`Any]};
```
Comment (samoht) [inline lib/block.ml]: why did you change that type?

File: lib/block.ml [hunk 2]
```diff
@@ -272,13 +273,23 @@ let environment t = match get_label t "env" with
   | Some (Some (`Eq, s)) -> s
   | Some (Some _) -> Fmt.failwith "invalid `env` label value"
 
-let variables t =
+(* ([ `Eq | `Ge | `Gt | `Le | `Lt | `Neq ] * string) option * string *)
```
Comment (samoht) [inline lib/block.ml]: you can remove that comment

File: lib/migrate_ast.ml [hunk 3]
```diff
@@ -32,7 +32,7 @@ module Parsetree_ = Parsetree
 
 include (
   Migrate_parsetree.Ast_406 :
-  module type of struct
+    module type of struct
```
Comment (samoht) [inline lib/migrate_ast.ml]: why did this file changes? can you clean-up the history and remove the merge commit?

**Comment by samoht** [comment]
Looks good with a few minor remarks.

Could you check that this doesn't break existing mdx users, for instance merlin or rwo?

File: lib/block.ml (hunk 1 above)
Comment (clecat) [reply lib/block.ml]: It felt clearer and cleaner, however the tags could be more explicit

File: lib/block.ml (hunk 2 above)
Comment (clecat) [reply lib/block.ml]: Indeed, my bad

File: lib/migrate_ast.ml (hunk 3 above)
Comment (clecat) [reply lib/migrate_ast.ml]: Indeed, I was late with master, that's why, i'll do the history rework

File: lib/block.ml (hunk 1 above)
Comment (samoht) [reply lib/block.ml]: Can you revert this change (and make an other PR if you really like the new record, even if I am not very fond of it..)  that'll make it easier to review.

Comment (clecat) [reply lib/block.ml]: Okay, no problem

**Comment by samoht** [comment]
Looks great, thanks!
